v1/contracts/UnsignedFixedPoint.sol

```diff
@@ -0,0 +1,44 @@
+pragma solidity ^0.5.0;
+
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+
+/**
+ * @title Library for fixed point arithmetic on uints
+ */
+library UnsignedFixedPoint {
+
+    using SafeMath for uint;
+
+    // Supports 18 decimals. E.g., 1e18 represents "1", 5e17 represents "0.5".
+    uint private constant FP_SCALING_FACTOR = 10**18;
+
+    struct FixedPointUint {
+        uint value;
+    }
+
+    function add(FixedPointUint memory a, FixedPointUint memory b) internal pure returns (FixedPointUint memory) {
+        return FixedPointUint(a.value.add(b.value));
+    }
+
+    function sub(FixedPointUint memory a, FixedPointUint memory b) internal pure returns (FixedPointUint memory) {
+        return FixedPointUint(a.value.sub(b.value));
+    }
+
+    function mul(FixedPointUint memory a, FixedPointUint memory b) internal pure returns (FixedPointUint memory) {
+        // There are two caveats with this computation:
+        // 1. Max output is 2^238 (i.e., 2^(256-18)), otherwise an intermediate value overflows.
```

Comment:
tiny nit: isn't the max value actually `(2^238) - 1` because the max value of a 256 bit uint is `(2^256)-1`?

Reply:
Actually, I think what I had there was wrong, I was mixing up powers of ten and powers of two :(

I updated the comments, can you check over my math?